Improve `expect_error` feedback

Blind4Basics opened the pull request:
previously:

when execting a specific error, if any other error is ancountered, the test fails without any useful feedback
suggestion:

added {actual error} should be {expected error} to the message
note:

this time, I updated the related test output fixture. I checked manually that the behavior was exactly the same (hopefuly... x) )

kazk commented:
You don't need to keep opening new PR to fix. Also, you probably need to update the test code to escape parentheses in `expected`. https://github.com/codewars/python-test-framework/blob/a07c83f0b5a0692f665ee316865510867ab9a4cf/tests/test_outputs.py#L23-L26

Blind4Basics commented:
\o/

finally passing...

Blind4Basics commented:
damn it... (I did another change...) I'll check again.

Blind4Basics commented:
good again...

@kazk: I tried to setup the thing to try it locally, but I ended up with the `result` of the subprocess being an empty string. No idea why... I had to modify the environment `Path`  object to a string too, otherwise I got errors too. No idea what are causing those differencies... (running 3.8.10 an Win7)

cheers

kazk commented:
Can you explain what you did (step by step) and what's failing? Screenshots and/or error messages helps.

The tests are passing here, so I'm not sure what's wrong.

Blind4Basics commented:
1. I pasted the two subfolders `codewars_test` and `tests` locally
2. updated the code
3. ran `tests/test_outputs.py`

there, I got this kind of error (same kind of message for every test method):

```python
======================================================================
ERROR: test_timeout_passing (__main__.TestOutputs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "...\codewars\CW - utilities\essais CW framework\tests\test_outputs.py", line 18, in test
    result = subprocess.run(
  File "...Programs\Python\Python38\lib\subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "...Programs\Python\Python38\lib\subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "...Programs\Python\Python38\lib\subprocess.py", line 1311, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
TypeError: environment can only contain strings
```

then I changed this:

```python
# from:
test_func = test_against_expected(
                os.path.join(fixtures_dir, f),
                expected_file,
                {"PYTHONPATH": package_dir},
            )
# to:
test_func = test_against_expected(
                os.path.join(fixtures_dir, f),
                expected_file,
                {"PYTHONPATH": str(package_dir)},
            )
```

There, the tests are running, but I get this:
```python
======================================================================
FAIL: test_timeout_passing (__main__.TestOutputs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "...\codewars\CW - utilities\essais CW framework\tests\test_outputs.py", line 32, in test
    self.assertRegex(result.stdout.decode("utf-8"), expected)
AssertionError: Regex didn't match: '\n<DESCRIBE::>group 1\n\n<PASSED::>Test Passed\n\n<PASSED::>Test Passed\n\n<COMPLETEDIN::>\\d+(?:\\.\\d+)?\n' not found in ''
```

kazk commented:
Confirmed that this is a Windows thing (#17). I don't have Windows, and we don't use Windows, so I'll leave it as is. If you find a fix, please open a PR.

kazk commented on tests/test_outputs.py:
```diff
@@ -20,9 +20,11 @@ def test(self):
         )
         with open(expected_file, "r", encoding="utf-8") as r:
             # Allow duration to change
+            expected = re.sub( r"([()])", r"\\\1", r.read() )
```
```suggestion
            expected = re.sub(r"([()])", r"\\\1", r.read())
```

kazk commented:
@Blind4Basics Deployed. Thanks!
